src/decoder/mod.rs

```diff
@@ -95,13 +126,36 @@ struct StripDecodeState {
     strip_bytes: Vec<u32>,
 }
 
+/// Decoding limits
+#[derive(Clone, Debug)]
+pub struct Limits {
+    /// The maximum size of any `DecodingResult` in bytes, the default is 
+    /// 256MiB. If the entire image is decoded at once, then this will
+    /// be the maximum size of the image. If it is decoded one strip at a
+    /// time, this will be the maximum size of a strip.
+    decoding_buffer_size: u64,
+    /// The maximum size of any ifd value in bytes, the default is 
+    /// 1MiB.
+    ifd_value_size: u64,
```

Comment:
Any particular reason why this is `u64` and not `usize`? You cannot really have a u64-sized buffer on a 32-bit platform.

Reply:
Not really, I just took from the `Limits` struct in the `png` crate, but you are right `usize` would be a better fit.